xarray/tests/test_coding_times.py

```diff
@@ -137,25 +143,23 @@ def test_cf_datetime(
         actual = decode_cf_datetime(num_dates, units, calendar, time_unit=time_unit)
 
     if actual.dtype.kind != "O":
-        expected = cftime_to_nptime(expected, time_unit=time_unit)
-
-    abs_diff = np.asarray(abs(actual - expected)).ravel()
-    abs_diff = pd.to_timedelta(abs_diff.tolist()).to_numpy()
+        if np.timedelta64(1, time_unit) > np.timedelta64(1, minimum_resolution):
+            expected_unit = minimum_resolution
+        else:
+            expected_unit = time_unit
+        expected = cftime_to_nptime(expected, time_unit=expected_unit)
```

Comment:
I needed to add this logic specifically for this test:
```
   (12300 + np.arange(5), "hour since 1680-01-01 00:00:00.500000", "us"),
```
This is because the reference date required the dates be decoded to microsecond resolution (really only millisecond resolution would be required, but we follow pandas's lead here).  Otherwise for `time_unit="s"` we would truncate precision when converting the cftime objects to `np.datetime64` values, which prevented asserting exact equality.